controllers/host/host_controller.go

```diff
@@ -2269,21 +2288,49 @@ func (r *HostReconciler) Reconcile(ctx context.Context, request ctrl.Request) (r
 	return ctrl.Result{}, nil
 }
 
-// SetupWithManager sets up the controller with the Manager.
-func (r *HostReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	tMgr := cloudManager.GetInstance(mgr)
-	r.Client = mgr.GetClient()
-	r.Scheme = mgr.GetScheme()
-	r.CloudManager = tMgr
-	r.ReconcilerErrorHandler = &common.ErrorHandler{
-		CloudManager: tMgr,
-		Logger:       logHost}
-	r.ReconcilerEventLogger = &common.EventLogger{
-		EventRecorder: mgr.GetEventRecorderFor(HostControllerName),
-		Logger:        logHost}
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&starlingxv1.Host{}).
-		Complete(r)
+// PlatformNetworkUpdateRequired checks and returns true if any of the
+// platform networks / address pools are out of sync.
+func (r *HostReconciler) PlatformNetworkUpdateRequired(instance *starlingxv1.Host) (error, bool) {
+	opts := client.ListOptions{}
+	opts.Namespace = instance.Namespace
+	platform_networks := &starlingxv1.PlatformNetworkList{}
+	err := r.List(context.TODO(), platform_networks, &opts)
+	if err != nil {
+		err = perrors.Wrap(err, "failed to list platform networks")
+		return err, false
+	}
+
+	for _, platform_network := range platform_networks.Items {
```

Comment:
Please modularize the section below to avoid nested loops

Reply:
Done.